### ISSUE TITLE
关于dive_into_keras_vgg16的预测脚本问题

### ISSUE THREAD

**Issue by guohuifengby** [HadXu/machine-learning#2]
你好，目录下的ex.py文件可以跑过，但是后面预测的那部分源码在哪里，感觉看的不是很懂。能否放上完整的脚本供参考，非常感谢。

## 现在我们开始来预测了 ##

首先写一个方法来加载并处理图片

```python
def load_image(imageurl):
    im = cv2.resize(cv2.imread(imageurl),(224,224)).astype(np.float32)
    im[:,:,0] -= 103.939
    im[:,:,1] -= 116.779
    im[:,:,2] -= 123.68
    im = im.transpose((2,0,1))
    im = np.expand_dims(im,axis=0)
    return im
```

## 读取vgg16的类别文件 ##

```python
f = open('synset_words.txt','r')
lines = f.readlines()
f.close()
```

```python
def predict(url):
    im = load_image(url)
    pre = np.argmax(model.predict(im))
    print lines[pre]
```

```python
%pylab inline
```

    Populating the interactive namespace from numpy and matplotlib
    

```python
from IPython.display import Image
```

```python
Image('cat1.jpg')
```

**Comment by Ontheroad123** [comment]
def load_image(url):
    im = cv2.resize(cv2.imread(url), (224, 224)).astype(np.float32)
    im[:,:,0] -= 103.939
    im[:,:,1] -= 116.779
    im[:,:,2] -= 123.68
    im = im.transpose((2,0,1))
    im = np.expand_dims(im, axis=0)
    return im


def predict(url):
    # 预测
    f = open('synset_words.txt', 'r')
    lines = f.readlines()
    im = load_image(url)
    pre = np.argmax(model.predict(im))
    print(lines[pre])
    f.close()

if __name__ == "__main__":

    # Test pretrained model
    model = VGG_16('vgg16_weights.h5')
    sgd = SGD(lr=1e-6, decay=1e-6, momentum=0.9, nesterov=True)
    model.compile(optimizer=sgd, loss='categorical_crossentropy')
    predict('cat.1.jpg')
